api_server/swagger/firecracker.yaml

```diff
@@ -501,6 +528,21 @@ definitions:
         type: string
         description: Host level path for the guest drive
 
+  PartialNetworkInterface:
```

Comment:
Pff, I remember doing this for `/drives` and deciding that we don't want to maintain 2 different structs that work for the same resource. We also didn't want to make everything an `Option` and go through validation hell. The least painful, but not ideal, way, was to deserialize any json for `PATCH` into a `Value` and force it to fit.

I don't like this. I think I'd rather rename `PatchDrivePayload` into `PatchPayload` and parse the `Value`, but I'm conflicted as this is a more complex use case. The thing is, if we end up supporting 2 structs for `network-interfaces`, we should do the same for `drives`, and everything `PATCH`-able... so I'm inclined to not do that.

Reply:
I left a comment under `vmm_config/mod.rs` about these structs. Basically, maintaining dumb/stateless structs is easier than maintaining custom/broken serialization code.